File: branches/Andean/Fantasy.Base/System.Threading/ThreadExtension.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace System.Threading
{
    public static class ThreadExtension
    {
        public static Thread WithStart(this Thread thread)
        {
            thread.Start();
            return thread;
        }

        public static Thread WithStart(this Thread thread, object state)
        {
            thread.Start(state);
            return thread;
        }
    }
}

File: AutoConverted/Fantasy.Base/System.Threading/ThreadExtension.java
package System.Threading;

public final class ThreadExtension
{
//C# TO JAVA CONVERTER TODO TASK: Extension methods are not available in Java:
//ORIGINAL LINE: public static Thread WithStart(this Thread thread)
	public static Thread WithStart(Thread thread)
	{
		thread.start();
		return thread;
	}

//C# TO JAVA CONVERTER TODO TASK: Extension methods are not available in Java:
//ORIGINAL LINE: public static Thread WithStart(this Thread thread, object state)
	public static Thread WithStart(Thread thread, Object state)
	{
		thread.Start(state);
		return thread;
	}
}